great_expectations/data_context/data_context.py

```diff
@@ -110,83 +93,124 @@ def create(cls, project_root_dir=None):
-            from great_expectations.jupyter_ux.expectation_explorer import ExpectationExplorer
-            self._expectation_explorer_manager = ExpectationExplorer()
-
-        # determine the "context root directory" - this is the parent of "great_expectations" dir
-        if context_root_dir is None:
-            if os.path.isdir("../notebooks") and os.path.isfile("../great_expectations.yml"):
-                context_root_dir = "../"
-            elif os.path.isdir("./great_expectations") and \
-                    os.path.isfile("./great_expectations/great_expectations.yml"):
-                context_root_dir = "./great_expectations"
-            elif os.path.isdir("./") and os.path.isfile("./great_expectations.yml"):
-                context_root_dir = "./"
-            else:
-                raise DataContextError(
-                    "Unable to locate context root directory. Please provide a directory name."
-                )
+        assert isinstance(project_config, DataContextConfig)
 
+        self._project_config = project_config
         self._context_root_directory = os.path.abspath(context_root_dir)
 
-        # TODO: these paths should be configurable
-        self.fixtures_validations_directory = os.path.join(self.root_directory, "fixtures/validations")
-        self.data_doc_directory = os.path.join(self.root_directory, "uncommitted/documentation")
-
-        self._project_config = self._load_project_config()
+        self._datasources = {}
 
         if not self._project_config.get("datasources"):
             self._project_config["datasources"] = {}
         for datasource in self._project_config["datasources"].keys():
             self.get_datasource(datasource)
 
-        plugins_directory = self._project_config.get("plugins_directory", "plugins/")
+        # self._plugins_directory = self._project_config.get("plugins_directory", "plugins/")
+        plugins_directory = self._project_config.get("plugins_directory")
+        #TODO: This check should not be necessary
+        if plugins_directory == None:
+            plugins_directory =  "plugins/"
+        #TODO: This should be factored out into a _convert_to_absolute_path_for_internal_use method, which should live in the @property, not here
         if not os.path.isabs(plugins_directory):
             self._plugins_directory = os.path.join(self.root_directory, plugins_directory)
         else:
             self._plugins_directory = plugins_directory
         sys.path.append(self._plugins_directory)
 
+        self._stores = DotDict()
+        #TODO: This check should not be necessary
+        if "stores" in self._project_config:
+            self._init_stores(self._project_config["stores"])
+
+
+        # Stuff below this comment is legacy code, not yet fully converted to new-style Stores.
+        self.data_doc_directory = os.path.join(self.root_directory, "uncommitted/documentation")
+
+        #TODO: Decide if this is part of the config spec or not
```

Comment:
Good question.

The issue was that `Expectations` act like a store (get and set via key lookup), but they're special in some ways. Should Expectations be treated like a Store? Or should they be something else?

After more reflection, I think the right approach is to do BOTH by changing the config parameter to `expectation_store_name`. This will point to a Store, configured just like any other Store. It will also provide a unique, non-optional "pointer" to expectations that can be used in methods like `DataContext.get_expectation_suite`.

This will support all current functionality, but also free us up to use things besides the local filesystem as the source of record for Expectations.

Reply:
...so in sum -- the TODO should be removed because the notion of an expectation store is definitely part of the datacontext as before.

The store underlying semantics are something we're about to come to in one of the later PRs I think.